cpp/src/arrow/compute/registry.cc

```diff
@@ -188,6 +188,9 @@ static std::unique_ptr<FunctionRegistry> CreateBuiltInRegistry() {
   RegisterScalarAggregateTDigest(registry.get());
   RegisterScalarAggregateVariance(registry.get());
 
+  // Random generator function
+  RegisterRandom(registry.get());
+
   RegisterAggregateOptions(registry.get());
```

Comment:
Move RegisterRandom after RegisterAggregateOptions.

Reply:
Shouldn't we group function registrations together? `RegisterAggregateOptions` is in a different category since it adds function options, not functions.

Reply:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/registry.cc#L158-L191
```
// scalar
register scalar kernels
register scalar option

// vector
register vector kernels
register vector option

// aggregate
register agg kernels
register agg option
```

Looks nullary kernel doesn't fit any existing type. We can add a new type at the end. E.g.,
```
// nullary
RegisterNullaryRandom
...
```

Reply:
Done.